### PR TITLE
chore: Update Discord username

### PR DESCRIPTION
**Pull request by adamdmharvey**
## Hey, I just made a Pull Request!

Minor documentation tweak given the Discord user name rename process rolling out.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: OWNERS.md [hunk 1]
```diff
@@ -120,7 +120,7 @@ Scope: Tooling for frontend and backend schema-first OpenAPI development.
 
 | Name                           | Organization              | GitHub                                                | Discord                        |
 | ------------------------------ | ------------------------- | ----------------------------------------------------- | ------------------------------ |
-| Adam Harvey                    | Cisco                     | [adamdmharvey](https://github.com/adamdmharvey)       | `adamharvey#3739`              |
+| Adam Harvey                    | Cisco                     | [adamdmharvey](https://github.com/adamdmharvey)       | `adamharvey_`                  |
 | Alex Crome                     |                           | [afscrome](https://github.com/afscrome)               | `afscrome`                     |
 | Andre Wanlin                   | Keyloop                   | [awanlin](https://github.com/awanlin)                 | `Ahhhndre#3095`                |
```
Comment (awanlin) [inline OWNERS.md:125]: Would you be open to updating mine while you are at it? New handle is just `ahhhndre`

Comment (adamdmharvey) [reply OWNERS.md:125]: @awanlin ; sure.  fire a suggestion and i'll commit it, so it'll give you credit on the commit as a co-author!

Comment (awanlin) [inline OWNERS.md:125]:
```suggestion
| Andre Wanlin                   | Keyloop                   | [awanlin](https://github.com/awanlin)                 | `ahhhndre`                |
```

Comment (awanlin) [reply OWNERS.md:125]: Thanks @adamdmharvey here's the change

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
Just fixed the prettier - should automerge when complete :tada:

**Pull request merged**